added example with cmake module inside tool_requires (#3291)
conan-io/docs: examples/graph/tool_requires/use_cmake_modules.rst

- The consumer wants to have the dependency cmake modules automatically loaded when the ``find_package()`` is executed. This current example implements this case.

Let's say that we have a package, intended to be used as a ``tool_require``, with the following recipe:

.. code-block:: python
    :caption: myfunctions/conanfile.py

    import os
    from conan import ConanFile
    from conan.tools.files import copy

    class Conan(ConanFile):
        name = "myfunctions"
        version = "1.0"
        exports_sources = ["*.cmake"]

        def package(self):
            copy(self, "*.cmake", self.source_folder, self.package_folder)

        def package_info(self):
            self.cpp_info.set_property("cmake_build_modules", ["myfunction.cmake"])

And a ``myfunction.cmake`` file in:

.. code-block:: cmake
    :caption: myfunctions/myfunction.cmake

    function(myfunction)
        message("Hello myfunction!!!!")
    endfunction()

> We can do a ``cd myfunctions && conan create .`` and that will create the ``myfunctions/1.0`` package containing the cmake script.

Reviewer: ```suggestion
We can do a ``cd myfunctions && conan create .`` which will create the ``myfunctions/1.0`` package containing the cmake script.
```

> Then, a consumer package will be like:

Reviewer: ```suggestion
Then, a consumer package will look like:
```
